### ISSUE TITLE
zapslog: Handler should inline the Attrs of a group with an empty key

### ISSUE THREAD

**Issue by arukiidou** [uber-go/zap#1402]
- part of #1334
  - may fixed by #1263

**Describe the bug**

`slogtest` cause errors:

```
missing key "c": a Handler should inline the Attrs of a group with an empty key (zap/exp/zapslog/slogtest.go:126)
```

actual logs:

```
zap\exp\zapslog\logger.go:130: 2024-01-28T01:08:54.902+0900	INFO	msg	{"a": "b", "": {"c": "d"}, "e": "f"}
```
**To Reproduce**
run `testing/slogtest`

```go
f: func(l *slog.Logger) {
	l.Info("msg", "a", "b", slog.Group("", slog.String("c", "d")), "e", "f")
},
```

**Expected behavior**

so that:

```
exp\zapslog\logger.go:130: 2024-01-28T01:08:54.902+0900	INFO	msg	{"a": "b", "c": "d", "e": "f"}
```

**Additional context**
none.

**Comment by arukiidou** [comment]
```go
	case slog.KindGroup:
		return zap.Object(attr.Key, groupObject(attr.Value.Group()))
```
to
```go
	case slog.KindGroup:
		if attr.Key == "" {
			return zap.Inline(groupObject(attr.Value.Group()))
		}
		return zap.Object(attr.Key, groupObject(attr.Value.Group()))
```

**Comment by abhinav** [comment]
Resolved by #1408
Thanks, @arukiidou
